### PR TITLE
bar_2

### LINKED ISSUE #151: bar_2
Description: Making a bar graph with bars above and below a reference line filled with a particular color

Script URL: https://www.ncl.ucar.edu/Applications/Scripts/bar_2.ncl
Plot URL: https://www.ncl.ucar.edu/Applications/Images/bar_2_lg.png

Category: Bar

### PR DESCRIPTION
**Pull request by hCraker**
Closes #151 
image

### CODE REVIEW

**Comment by hCraker** [comment]
I'm having issues with the bars overlapping. I've tried playing with the `width` kwarg, but I keep getting unexpected results. Reducing the width will result in some of the bars being right next to each other and others with space in between them. @clairefio @michaelavs Have either of encountered something like this?

**Comment by hCraker** [comment]
This is better. There's only a bit of overlap and it's hard to notice at first.
image

File: Plots/Bar/NCL_bar_2.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+###############################################################################
+# Plot
+
+# Generate figure (set its size (width, height) in inches) and axes
+plt.figure(figsize=(12, 6))
+ax = plt.axes()
+
+# Create a list of colors based on the color bar values
+colors = ['red' if (value > 0) else 'blue' for value in dsoik[::8]]
+plt.bar(date_frac[::8], dsoik[::8], align='edge', edgecolor='black', color=colors, width=8/12)
```
Comment (clairefio) [inline Plots/Bar/NCL_bar_2.py:54]: Make the bar edges thinner to match the NCL graph
```suggestion
plt.bar(date_frac[::8], dsoik[::8], align='edge', edgecolor='black', color=colors, width=8/12, linewidth=.6)
```

**Comment by michaelavs** [comment]
Looks good so far! I think that you will really only be able to change the line width and width of the bars to prevent overlap, so may mess around with that a bit more, but if you're happy with where it is now, then I think it's fine too.

**Comment by hCraker** [comment]
image

**Comment by hCraker** [comment]
I accidentally pushed changes to this repository that were meant for another branch and pull request. That's what the force push was fixing.

**Comment by jukent** [comment]
I'm not seeing the overlap. Could you circle it?

**Comment by hCraker** [comment]
> I'm not seeing the overlap. Could you circle it?

I'm beginning to think that it is an artifact caused by my old monitor's low resolution. The group of 4 red bars just before 1920 always look like they have different widths on my display.

**Comment by jukent** [comment]
> > I'm not seeing the overlap. Could you circle it?
> 
> I'm beginning to think that it is an artifact caused by my old monitor's low resolution. The group of 4 red bars just before 1920 always look like they have different widths on my display.

Oh I see it in the old plot. It looks fixed now.

**Pull request merged**